bet/postProcess/compareP.py

```
@@ -1,1080 +1,349 @@
+
+        :param set1: Object containing left probability measure.
+        :type set1: :class:`bet.sample.sample_set` or `bet.sample.discretization`lass:
+        :param set2: Object containing left probability measure.
+        :type set1: :class:`bet.sample.sample_set` or class:`bet.sample.discretization`
+        :param inputs: If set1 and set2 are discretizations, use input sets if True and output if False.
+            True by default.
+        :type inputs: bool
+        :param set1_init: Use initial probability measure for set1 if True. False by default.
+        :type set1_init: bool
+        :param set2_init: Use initial probability measure for set2 if True. False by default.
+        :type set2_init: bool
+        """
+        self.pdfs1 = None
+        self.pdfs2 = None
+        self.compare_vals = None
+        self.set1_init = set1_init
+        self.set2_init = set2_init
+        self.pdfs_zero = None
+
+        # Extract sample sets
+        if isinstance(set1, samp.discretization):
+            if inputs:
+                set1 = set1.get_input_sample_set()
             else:
-                self._domain = sample_set_right.get_domain()
+                set1 = set1.get_output_sample_set()
 
-        # check dimension consistency
-        if isinstance(comparison_sample_set, samp.sample_set_base):
-            self._num_samples = comparison_sample_set.check_num()
-            output_dims = []
-            output_dims.append(comparison_sample_set.get_dim())
-            if self._right_sample_set is not None:
-                output_dims.append(self._right_sample_set.get_dim())
-            if self._left_sample_set is not None:
-                output_dims.append(self._left_sample_set.get_dim())
-            if len(output_dims) == 1:
-                self._comparison_sample_set = comparison_sample_set
-            elif np.all(np.array(output_dims) == output_dims[0]):
-                self._comparison_sample_set = comparison_sample_set
```

I was making such checks to keep in spirit with how I saw BET was coded before.

However, boy does it suck to read. Yeah, user-error is less protected against, but as long as it fails in a way that makes sense for debugging, I'm in favor of removal / moving forward with less hand-holding / error-prevention. I want things to fail fast if we're going to skip this stuff. 

Do we have tests to ensure failing cases are handled properly? That was something I remember doing. Is it something we should bother with at all going forward?